### PR TITLE
Remove CSI Node related funcs from cache

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
From discussion in #79416, it seems determining the proper order of Node / CSINode addition / removal is non-trivial.

This PR removes CSI Node related funcs from cache.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @bsalamat @bertinatto @msau42

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -157,59 +157,39 @@ func (sched *Scheduler) deleteNodeFromCache(obj interface{}) {
...
 	if !ok {
 		klog.Errorf("cannot convert to *storagev1beta1.CSINode: %v", obj)
 		return
 	}
 
-	if err := sched.config.SchedulerCache.AddCSINode(csiNode); err != nil {
-		klog.Errorf("scheduler cache AddCSINode failed: %v", err)
-	}
-
 	sched.config.SchedulingQueue.MoveAllToActiveQueue()
```
Comment (bertinatto) [inline pkg/scheduler/eventhandlers.go]: I don't think this is correct. The CSINode isn't set at all, so there will be no volume limits.

Comment (tedyu) [reply pkg/scheduler/eventhandlers.go]: @bertinatto 
Do you have recommendation on what onCSINodeUpdate should do ?

Comment (tedyu) [reply pkg/scheduler/eventhandlers.go]: The SchedulerCache.AddCSINode call may return error (which was ignored).
It seems this func doesn't depend on the successful addition to cache.

**Pull request closed without merging**